File: week1/5-Sorting/selection_sort.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

class SelectionSort
{
    public:
        void sort(int*, size_t);
};

void SelectionSort::sort(int* sequence, size_t size)
{
    for (size_t i = 0; i < size - 1; ++i)
    {
        size_t min_index = i;

        for (size_t j = i + 1; j < size; j++)
        {
            if (sequence[min_index] > sequence[j])
            {
                min_index = j;
            }
        }

        if (sequence[i] != sequence[min_index])
        {
            int temp = sequence[i];
            sequence[i] = sequence[min_index];
            sequence[min_index] = temp;
        }
    }
}

File: week1/5-Sorting/selection_sort.py
class SelectionSort:

    def selection_sort(sequence):
        """Sorts a sequence of integers"""

        for i in range(len(sequence)):
            subseq = sequence[i:]
            min_element = min(subseq)
            min_index = subseq.index(min_element) + i

            sequence[i], sequence[min_index] = sequence[min_index], sequence[i]

        return sequence


